sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -427,6 +435,26 @@ private CarbonFileInputFormat prepareFileInputFormat(Job job, boolean enableBloc
     }
   }
 
+  private <T> void totalRowCountInSplits(Job job, List<InputSplit> splits,
+      List<Long> rowCountInSplit) throws IOException, InterruptedException {
+    long sum = 0;
+    for (InputSplit split : splits) {
+      List<RecordReader<Void, T>> readers = new ArrayList<>();
```

Comment:
please try to check if the filter is present, then prepare a reader and get the split count like you did now.
Else handle it like previous (by just reading total rows in a split), because reading rows again is kind of slow.

Reply:
If store has delta files || filter is present --> use new way by building reader and get the count. Else do old way.

Reply:
done